schemas/iso19115-3.2018/src/main/plugin/iso19115-3.2018/index-fields/link-utility.xsl

```diff
@@ -104,37 +104,39 @@
                     select="current()"/>
       <xsl:for-each select="$root/descendant::*[
                               local-name() = $docType/element/text()
-                              ]/*[cit:onlineResource/*/cit:linkage/gco:CharacterString != '']">
+                              ]/*/cit:onlineResource/*[cit:linkage/gco:CharacterString != '']">
         <item>
           <id>
-            <xsl:value-of select="cit:onlineResource/cit:CI_OnlineResource/cit:linkage/gco:CharacterString"/>
+            <xsl:value-of select="cit:linkage/gco:CharacterString"/>
           </id>
           <url>
             <xsl:choose>
               <xsl:when test="$forIndexing">
-                <xsl:copy-of select="cit:onlineResource/*/cit:linkage"/>
+                <xsl:copy-of select="cit:linkage"/>
               </xsl:when>
               <xsl:otherwise>
                 <xsl:apply-templates mode="get-iso19115-3.2018-localized-string"
-                                     select="cit:onlineResource/*/cit:linkage"/>
+                                     select="cit:linkage"/>
               </xsl:otherwise>
             </xsl:choose>
           </url>
           <title>
+            <xsl:variable name="name"
+                          select="if (cit:name) then cit:name else ../../cit:title"/>
             <xsl:choose>
               <xsl:when test="$forIndexing">
-                <xsl:copy-of select="cit:title"/>
+                <xsl:copy-of select="$name"/>
               </xsl:when>
               <xsl:otherwise>
                 <xsl:apply-templates mode="get-iso19115-3.2018-localized-string"
-                                     select="cit:title"/>
+                                     select="$name"/>
               </xsl:otherwise>
             </xsl:choose>
           </title>
           <description>
             <xsl:variable name="desc"
-                          select="if (cit:onlineResource/*/cit:description)
-                                        then cit:onlineResource/*/cit:description
+                          select="if (cit:description)
+                                        then cit:description
                                         else ../../mdq:abstract"/>
```

Comment:
This xpath doesn't seem to be fine, I think should be `../../../../mdq:abstract`.

The loop xpath is `//mdq:reportReference/cit:CI_Citation/cit:onlineResource/cit:CI_OnlineResource`, `../../mdq:abstract` can access the `mdq:abstract` element:

```
<mdq:reportReference>
    <cit:CI_Citation>
        <cit:onlineResource>
            <cit:CI_OnlineResource>
                <cit:linkage>
                    <gco:CharacterString>http://www.osi-saf.org/</gco:CharacterString>
                </cit:linkage>
                <cit:protocol>
                    <gco:CharacterString>WWW:LINK</gco:CharacterString>
                </cit:protocol>
                <cit:name>
                    <gco:CharacterString>Radiative fluxes over Indian Ocean from METEOSAT-8 data, validation report</gco:CharacterString>
                </cit:name>
                <cit:description>
                    <gco:CharacterString>Validation report for OSI SAF Meteosat Radiatives fluxes</gco:CharacterString>
                </cit:description>
                <cit:function>
                    <cit:CI_OnLineFunctionCode codeList="http://standards.iso.org/iso/19115/resources/Codelists/cat/codelists.xml#CI_OnLineFunctionCode"
                        codeListValue="information"/>
                </cit:function>
            </cit:CI_OnlineResource>
        </cit:onlineResource>
        ...
    </cit:CI_Citation>
</mdq:reportReference>
<mdq:abstract>
    <gco:CharacterString>Report abstract</gco:CharacterString>
</mdq:abstract>
```